### DISCUSSION TITLE
Create index pages?

### DISCUSSION THREAD

**Discussion by peter88213** [peter88213/zim2obsidian#6] [Ideas]
The Zim Markdown export tree features the Zim wiki structure. 

### Exported structure (example):
```
leaf1.md
leaf2.md
node1.md
node1 <Dir>
          leaf3.md
          leaf4.md
          node2.md
          node2 <Dir>
                    leaf5.md
                    leaf6.md
          node3.md
          node3 <Dir>
                    leaf7.md
```

Should this structure be changed, so that the *nodeX.md* pages are moved
to the directory representing *nodeX*, and renamed *index.md*?

### Modified structure (example):
```
leaf1.md
leaf2.md
node1 <Dir>
          index.md
          leaf3.md
          leaf4.md
          node2 <Dir>
                    index.md
                    leaf5.md
                    leaf6.md
          node3 <Dir>
                    index.md
                    leaf7.md
```
All links to the former *nodeX.md* would have to be changed to *nodeX/index.md*. This part involves a certain risk of error.

**Comment by sojusnik** [comment]
Regardless of a [direct or indirect conversion](https://github.com/peter88213/zim2obsidian/discussions/8), this issue should be addressed anyway.

In my case a lot of `node1.md` notes from your example have content and I actively link to them, but after converting with your script, the corresponding links are linking to `node1 <Dir>` instead of `node1.md`, thus the linking in such cases isn't working, because Obsidian treats folders not as notes, but merely as an organizational element. So your suggestions seems reasonable and maybe will also solve the linking issue automatically after conversion.

The only workaround I can think of right now is to manually move all notes to the top level in Zim, so that no folders are present, make the conversion and only then structure the notes within Obsidian. But I guess this is will be too laborious for the majority of users.

**Reply by peter88213** [reply]
Another thing could be moving the nodeX.md pages one level up without renaming them. 

Yesterday I began an implementation just for fun,  with the [3-create-index-pages branch](https://github.com/peter88213/zim2obsidian/tree/3-create-index-pages). Moving the nodeX.md pages is easy, but adjusting the links turned out to be tricky. 

Because I'm very behind with my current novel right now, I'm not going to pursue this topic for the time being.

**Reply by peter88213** [reply]
> after converting with your script, the corresponding links are linking to `node1 <Dir>` instead of `node1.md`, thus the linking in such cases isn't working, because Obsidian treats folders not as notes, but merely as an organizational element.

This may be due to a bug to be fixed with v0.5.0.
